src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,37 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ handleFormSubmit, modalSelector, loadingText }) {
+    super({ modalSelector });
+    this._popupForm = this._modalElement.querySelector(".modal__form");
+    this._handleFormSubmit = handleFormSubmit;
+    this._loadingText = loadingText;
+    this._confirmDeleteButton = this._modalElement.querySelector(
+      "#modal-delete-button"
+    );
+    this._submitButton = this._modalElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._confirmDeleteButton.textContent = this._loadingText;
+    } else {
+      this._confirmDeleteButton.textContent = this._submitButtonText;
+    }
+  }
+
```

Comment:
The page should not reload when you delete a card. Please prevent the default submit event behaviour.

Reply:
Please prevent the default event behaviour in the submit handler

```js
_onSubmit = (event) => {
  event.preventDefault();
  this._handleFormSubmit()
}

// when opening popup
this._popupForm.addEventListener("submit", this.onSubmit);

// when closing popup
this._popupForm.removeEventListener("submit", this.onSubmit);
```